pkg/http/server.go

```diff
@@ -81,6 +90,43 @@ func (s *server) setupListener(opts Opts) error {
 	return nil
 }
 
+func setupUnixSocketListener(networkType string, address string) (net.Listener, error) {
+	socketOpts := strings.Split(address, ",")
```

Comment:
Technically this is a breaking change if someone's socket path previously contained a `,` character, which is legal on Linux at least.

I'm not sure if one would qualify that as "important note" or "breaking change" however in practice

Reply:
Why not have --http-address-mode as a separate parameter instead?

Reply:
As far as I can tell `,mode=` is not a convention when specifying sockets, so it might make sense to introduce an additional parameter indeed. We do have quite some configuration parameters already though, so I'm not sure if this is desired in that regard. @JoelSpeed @tuunit any thoughts/opinions?

Reply:
I mentioned it a bit in the issue itself (rather than the PR); mainly that I figured this is quite a niche feature and adding complexity to the configuration was perhaps a bit much.

Especially with the alpha configuration where the hierarchical approach strongly suggests we'd ought to convert from string to struct as it's a property of the listener, not of the server as a whole...

While it's not strictly a common approach, at the end of the day those kinds of suffixed parameters have precedents in popular software, notably Docker and Varnish, and that was "good enough" to me

Reply:
My point being that this does not break existing setups. The current parameter is http-<ADDRESS>, which is not the same as, say, --unix-path=/tmp/sock,mode=0660. I have never seen unix:// URI specify mode.

Reply:
I get that point, yes, but I am not sure this is right...

Changing uid, gid, mode, executable bits, etc, is not very common, but not without precedent.

For example the syntax for docker compose volumes (not sockets, but the idea remains):
```compose
services:
  foo:
    ...
    tmpfs:
      - /var/lib/foo:exec,uid=102,gid=103
```

or Varnish (here, sockets specifically):
```
/usr/sbin/varnishd ... \
  -a '/path/too/varnish.sock,user=foo,group=bar,mode=0644'
```

I am repeating myself with these examples, but this is not easy stuff to Google around for.

Extending the configuration schema or CLI flags for a feature maybe useful to <1% of users (if even that) seems a lot to me. If that is the preferred way, fine of course, but I would certainly be wary of it if I were in charge.

And by rare need, I mean that in this case the only reason I want to make it work that way is to have multi-tenancy by running multiple oauth2-proxy instances without needing to hog 2 ports per-instance forever (http+metrics), and that my reverse proxy runs chrooted with an unprivileged user that should have as few permissions as possible. Otherwise I'd long have moved on with simpler HTTP listeners. 

Also of note, as I hadn't thought about this before, I can probably make it work with unix domain sockets (ie `unix://@foo` binds successfully), as those aren't bound to a file, and thus carry no file permissions (I believe. Documentation is *sparse*). So maybe the solution is actually to not add the feature at all.

If the feature still makes sense, and backwards compatibility for the (imo theoretical) case of previous installs with a `,` in the socket parameters, I'd still suggest not making a dedicated CLI flag on --http* hierarchy, and instead:
1. note it in the changelog
2. gate that behind a sort of feature-flag section of the config, where users can opt into some breaking changes (which are to become default in a future major release); there's a couple other places that would benefit from that